### DISCUSSION TITLE
Git Extension Api Questions

### DISCUSSION THREAD

**Discussion by ZakaRehman** [microsoft/vscode-discussions#2316] [Extension Development QnA] [answered]
I am trying to get all commits done by a user, I am using the log function on the repository interface and when I supply an author, I get no commits, with no author supplied I get the commits done all authors. I have tried supplying the user name, the user email and the user name and user email matching what is shown if I run git log in the terminal and still I get no commits from the log function.

**Comment by isidorn** [comment]
What are you trying to achieve in your extension? Are you still facing this issue?

**Reply by ZakaRehman** [reply, marked as answer]
I was trying to build an extension that would fetch me a number of commits given an a author and then calculate the number of working days based on the days committed, I am a no longer facing this issue due to no longer using the git extension api and instead using simple-git, which is providing the necessary functionality to do what is needed

**Reply by isidorn** [reply]
Makes sense. Happy to hear you found a solution.
